### DISCUSSION TITLE
Deadline approaching

### DISCUSSION THREAD

**Discussion by mislavmatijevic** [foivz/pop-app#21] [General]
With the deadline approaching, I suggest we ditch NFC and just focus on QR code payments.

I've looked into the old QR generation and its awful. The QR code just contains the amount (for whatever reason) and it's multiplied by some abnormally huge number just to look good on the screen (QRCodeActivity.kt, `expand` method).

**Comment by zstapic** [comment]
This comment will enter into my teaching materials :)

[image: image.png]

**Reply by jfletcher20** [reply]
This means we can expect better from further generations, @mislavmatijevic!

**Comment by jfletcher20** [comment]
I concur, let's focus on the QR code payment system.

**Comment by jfletcher20** [comment]
Conclusion:
We were unable to build for iOS in time, so we ended up deploying the app to the cortext.foi.hr server and it worked perfectly except for QR code scanning - the camera wouldn't work for some reason - so we ended up additionally implementing payments via code input.

Overall a great success with no major issues on our part, the only issues were caused by the poorly-written backend.
